specs/chugsplash-config.md

```diff
@@ -21,32 +21,90 @@ ChugSplash config files may be written in Javascript, Typescript, or JSON. For J
+### OrganizationId
+A 32 byte string which identifies the organization that the relevant ChugSplash config is a part of. Each organization gets a unique ChugSplashManager contract which manages deployments for their organization. Users of the ChugSplash managed service have an organization id recommended to them, but other users can just generate and use any arbitrary 32 byte string.
 
-> Reference names are used internally by ChugSplash to identify specific instances of contracts. Each combination of a project name and contract reference name should be unique within the ChugSplash system.
+You can think of the organization id as identifying the toplevel address space of the users organization.
 
-## Contract Config Definition
-JSON or Javascript object defining a contract to be deployed, it's constructor args, and variable definitions. Contract definitions must conform to the following format:
+## Project Name
+An abitrary name string which identifies a specific project within an Organization. Project names are unique within each organization. If you create multiple configs in the same organization that use the same project name, then they will be treated like they are part of the same project when you perform deployments. We recommend that project names not be shared between multiple config files.
 
-```js
-ReferenceName: {
-  contract: 'LiteralContractName',
-  constructorArgs: {},
-  variables: {}
-}
-```
+You can think of the project name as identifying a subsection of the address space of the organization that the project is a part of. Project names allow ChugSplash to reuse the same ChugSplashManager within an organization while avoid the possibility of different projects deployments and upgrades impacting each other.
+
+## Claimer
+The address of the wallet used to claim the organization on chain. Since ChugSplash deterministically deploys contracts to the same addresses across chains, we have to account for the possibility that a malicious third party could attempt to snipe the contract addresses of well know protocols on chains they have not deployed to yet. To avoid this attack vector, we use the address of the claimer wallet as the final element in determining the address of the ChugSplash manager. This makes it impossible to snipe a ChugSplash users contract address without owning the contract used to claim the organization.
 
-## Contract Constructor Args
-A JSON or Javascript object defining a set of constructor args. All constructor arguements are required to be immutable variables...
+# Contracts Config Field
+JSON or Javascript object defining the contracts that should be deployed. Each key is a contracts reference name and its value is a contract definition.
 
-TODO finish defining constructor args spec
+> Reference names are used internally by ChugSplash to identify specific instances of contracts. Each combination of a project name and contract reference name should be unique within the ChugSplash system.
```

Comment:
> Reference names are used internally by ChugSplash to identify specific instances of contracts.

Maybe shorten this to: "Reference names identify specific instances of contracts in the ChugSplash system."

Since they're also used externally by users